api/orchestration_movers.go

```diff
@@ -452,3 +452,34 @@ func (o *datasyncOrchestrator) taskDetailsFromName(ctx context.Context, group, n
 
 	return nil, nil, apierror.New(apierror.ErrNotFound, "datasync mover not found", nil)
 }
+
+// taskRunsFromTaskArn finds a datasync task runs  based on its group/name and returns information about it
+func (o *datasyncOrchestrator) taskRunsFromTaskArn(ctx context.Context, taskArn string) ([]string, error) {
+	if taskArn == "" {
+		return nil, apierror.New(apierror.ErrBadRequest, "invalid input", nil)
+	}
+
+	// get a list of all datasync resources in the group
+	out, err := o.datasyncClient.ListDatasyncTasksexecutions(ctx, taskArn)
+	if err != nil {
+		return nil, err
+	}
+
+	//VISIT  : When 0 should return error or empty array
+	if len(out) == 0 {
+		return nil, apierror.New(apierror.ErrNotFound, "datasync mover not found", nil)
+	}
+
+	return out, apierror.New(apierror.ErrNotFound, "datasync mover not found", nil)
+}
+
+func (o *datasyncOrchestrator) TaskDetailsFromid(ctx context.Context, id string) (*datasync.DescribeTaskExecutionOutput, error) {
+	if id == "" {
+		return nil, apierror.New(apierror.ErrBadRequest, "invalid input", nil)
+	}
+	out, err := o.datasyncClient.DescribeTaskExecution(ctx, id)
+	if err != nil {
+		return nil, err
+	}
+	return out, apierror.New(apierror.ErrNotFound, "datasync mover not found", nil)
```

Comment:
should be `return out, nil`
also, we want to return our own custom struct here, e.g. `DatasyncRun` which only contains the fields we care about (similar to how we do it in the `datamoverDescribe` func

Reply:
Need more info on shape of the structure..

Reply:
Something like this based on https://docs.aws.amazon.com/sdk-for-go/api/service/datasync/#DescribeTaskExecutionOutput

```
type DatamoverRun struct {
	BytesTransferred *int64
	BytesWritten *int64
	EstimatedBytesToTransfer *int64
	EstimatedFilesToTransfer *int64
        FilesTransferred *int64
        StartTime *time.Time
        Status *string
        Result *DatamoverRunResult
}
```

and also another struct `DatamoverRunResult` that's basically a copy of https://docs.aws.amazon.com/sdk-for-go/api/service/datasync/#TaskExecutionResultDetail (I think we want all the fields)

Reply:
Added this in types.go and returned in by ID API call